### ISSUE TITLE
LayoutLoader can't find source layout if the project is inside a folder containing a dot in its name

### ISSUE THREAD

**Issue by hartbit** [schibsted/layout#70]
In `findProjectDirectory`, once the recursive function reaches the folder containing the project, if the name of the folder contains a dot (for example, as is the case if you download a tagged release from GitHub - `layout-0.5.7`), `pathExtension` will be non-empty and `findProjectDirectory` will incorrectly believe the last path component is a file and go one level up, never searching the folder containing the Xcode project.

PS: Sorry I couldn't find the time to write a PR, hopefully I provided enough detail for this to be easy to fix.

**Comment by hartbit** [comment]
Okay, here's a rewrite of `findProjectDirectory` which uses sequence generation instead of recursion and resolves the bug. I didn't yet open a PR because I didn't write any tests yet, but it seems to work fine:

```
    private func findProjectDirectory(at path: String) -> URL? {
        if let projectDirectory = _projectDirectory, path.hasPrefix(projectDirectory.path) {
            return projectDirectory
        }

        var url = URL(fileURLWithPath: path).standardizedFileURL
        if !url.hasDirectoryPath {
            url.deleteLastPathComponent()
        }

        let xcodePathExtensions = ["xcodeproj", "xcworkspace"]
        return sequence(first: url, next: { url in
            let url = url.deletingLastPathComponent()
            return !url.absoluteString.isEmpty ? url : nil
        }).first(where: { directoryURL in
            let files = try! FileManager.default.contentsOfDirectory(
                at: directoryURL,
                includingPropertiesForKeys: nil,
                options: [])
            return files.contains(where: { xcodePathExtensions.contains($0.pathExtension) })
        })
    }
```

**Comment by nicklockwood** [comment]
Great investigation, thanks!

**Comment by nicklockwood** [comment]
Fixed in 0.5.8
